demux/packet_pool.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+struct demux_packet_pool;
+struct demux_packet;
+struct mpv_global;
+
+/**
+ * Initializes the demux packet pool.
+ *
+ * This function creates a new shaderd demux packet pool. Should be done only
+ * once per mpv context.
+ *
+ * @param global Pointer to the global context.
+ */
+void demux_packet_pool_init(struct mpv_global *global);
+
+/**
+ * Uninitializes and cleans up the demux packet pool.
+ *
+ * This function destroys the mutex and clears the packet list by
+ * freeing all stored demux packets. This function is thread-safe.
+ *
+ * @param p Pointer to the demux packet pool.
+ */
+void demux_packet_pool_uninit(void *p);
+
+/**
+ * Returns the demux packet pool context for client use.
+ *
+ * @param global Pointer to the global context.
+ * @return Pointer to the demux packet context.
+ */
+struct demux_packet_pool *demux_packet_pool_get(struct mpv_global *global);
+
+/**
+ * Clears the demux packet pool.
+ *
+ * This function frees all the packets in the pool and clears the
+ * linked list while locking the pool to ensure thread safety.
```

Comment:
implementation detail 
```suggestion
 * This function frees all the packets in the pool.
```

Reply:
Done.